extensions/ql-vscode/src/pure/bqrs-utils.ts

```diff
@@ -97,20 +97,29 @@ export function isStringLoc(loc: UrlValue): loc is string {
 
 export function tryGetRemoteLocation(
   loc: UrlValue | undefined,
-  fileLinkPrefix: string
+  fileLinkPrefix: string,
+  sourceLocationPrefix: string,
 ): string | undefined {
   const resolvableLocation = tryGetResolvableLocation(loc);
   if (!resolvableLocation) {
     return undefined;
   }
 
-  // Remote locations have the following format:
-  // file:/home/runner/work/<repo>/<repo/relative/path/to/file
-  // So we need to drop the first 6 parts of the path.
+  let trimmedLocation: string;
 
-  // TODO: We can make this more robust to other path formats.
-  const locationParts = resolvableLocation.uri.split('/');
-  const trimmedLocation = locationParts.slice(6, locationParts.length).join('/');
+  // Remote locations have the following format:
+  // "file:${sourceLocationPrefix}/relative/path/to/file"
+  // So we need to strip off the first part to get the relative path.
+  if (sourceLocationPrefix) {
```

Comment:
Technically it could be undefined (for older queries that were run before the upstream change, ie before last week), but in that case the `if (...)` will still handle those as expected.